### ISSUE TITLE
Question about test/csl/sort_OmittedBibRefMixedNumericStyle.txt

### ISSUE THREAD

**Issue by jgm** [citation-style-language/test-suite#28]
Something I don't understand here.
The layout is
``` xml
      <choose>
        <if match="none" type="personal_communication">
          <group delimiter=". ">
            <text variable="citation-number"/>
            <group delimiter=", ">
              <text macro="author" />
              <text variable="title"/>
            </group>
          </group>
        </if>
      </choose>
```
I would assume  that this means that items with type `personal_communication` produce no output.  But oddly, the expected RESULT includes the citation number:
``` xml
<div class="csl-bib-body">
  <div class="csl-entry">1. Anderson, Book One</div>
  <div class="csl-entry">2. [CSL STYLE ERROR: reference with no printed form.]</div>
  <div class="csl-entry">3. Crane, Book Two</div>
</div>
```
See the `2.` in the second entry.  Why?

**Comment by bdarcus** [comment]
I agree that looks wrong, and that the 2 div shouldn't be there (right?).

Can @adam3smith or @fbennett confirm?

**Comment by adam3smith** [comment]
I think John's question isn't why the `<div>`  element is there, but why it contains the citation number, which is inside the if condition. 

I agree that logically this seems wrong. That said, I don't think the behavior for citations without a printed form is currently defined in the specs (right?) and I'm guessing Frank added the citation number for easier identification of the problematic item. I think to proceed we'd first want to specify the no-printed-form behavior (which exists to avoid terrible messes, such as items with invisible citations, bibliographies skipping citations numbers, etc.) and then decide if there should or shouldn't be a special rule for including citation numbers either way or not.

**Comment by bdarcus** [comment]
So we need language in the spec, ideally for 1.1, that addresses two cases:

1. nocite (but reference printed in bib)
2. the opposite (this one; cite, but no bib entry)

... and then to update this and any related test(s) accordingly?

Right?

**Comment by bwiernik** [comment]
Related issue: https://github.com/citation-style-language/documentation/issues/74

I think we would need to handle three cases:
1. `nocite`: uncited items that are included in the bibliography (this is pandoc's `nocite` or if a user used Zotero's Edit Bibliography function to add an uncited item to the bibliography)
2. "noref": cited items that are excluded from the bibliography using a conditional (this case). This should be allowed as many styles require it.
3. "citation with no visible form": A cited item for which the **citation** (not bibliography) is empty. This should return an error. An item appearing in a citation must have some visible output or it can lead to problematic behavior for users, especially for word processor-based applications. This is different from `nocite`.

**Comment by jgm** [comment]
I guess the idea must be that in a numeric style, it would be confusing to omit an item altogether, since then the numbers would not be consecutive. So this is handled by a special case, even though it goes against the strict letter of the spec.

**Comment by jgm** [comment]
But in my opinion a better solution would just be to write the style differently:
``` xml
        <group delimiter=". ">
          <text variable="citation-number"/>
           <choose>
             <if match="none" type="personal_communication">
               <group delimiter=", ">
                 <text macro="author" />
                 <text variable="title"/>
              </group>
        </if>
      </choose>
    </group>
```

**Comment by bwiernik** [comment]
@jgm See my comments here: https://github.com/citation-style-language/documentation/issues/92

**Comment by bwiernik** [comment]
> But in my opinion a better solution would just be to write the style differently:

I agree. Issues about processing noref citations aside, this test seems to be written wrong.
